src/main/java/org/takes/rq/RqMultipart.java

```diff
@@ -474,15 +450,26 @@ public InputStream body() throws IOException {
          * Fake body creator.
          * @param dispositions Fake request body parts
          * @return InputStream of given dispositions
+         * @throws IOException If fails
          */
         @SuppressWarnings("PMD.InsufficientStringBufferDeclaration")
-        private static InputStream fakeBody(final String... dispositions) {
+        private static InputStream fakeBody(final Request... dispositions)
+            throws IOException {
             final StringBuilder builder = new StringBuilder();
-            for (final String disposition : dispositions) {
-                builder.append(String.format("--%s%s", BOUNDARY, "\r\n"))
-                    .append(disposition)
+            for (final Request each : dispositions) {
+                builder.append(String.format("--%s%s", BOUNDARY, CRLF))
                     // @checkstyle MultipleStringLiteralsCheck (1 line)
-                    .append("\r\n");
+                    .append("Content-Disposition: ")
+                    .append(
+                        new RqHeaders.Smart(
+                            new RqHeaders.Base(each)
+                        // @checkstyle MultipleStringLiteralsCheck (1 line)
+                        ).single("Content-Disposition")
+                    ).append(CRLF);
+                final String body = new RqPrint(each).printBody();
+                if (!(CRLF.equals(body) || "".equals(body))) {
+                    builder.append(String.format("\r\n%s\r\n", body));
```

Comment:
@erimerturk maybe we could do this without formatting?
```
builder.append(CRLF).append(body().append(CRLF)
```
It's more natural "builder" way ... we will also reuse CRLF constant